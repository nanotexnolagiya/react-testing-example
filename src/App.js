import React from 'react';

import { BrowserRouter, Switch, Route } from 'react-router-dom';
import { Tasks, Posts } from './pages';
import { MainLayout } from './layouts/MainLayout';

const App = () => (
  <BrowserRouter>
    <MainLayout>
      <Switch>
        <Route path="/" exact component={Tasks} />
        <Route path="/posts" exact component={Posts} />
      </Switch>
    </MainLayout>
  </BrowserRouter>
);

export default App;