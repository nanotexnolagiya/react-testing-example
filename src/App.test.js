import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom'
import App from './App';

test('renders learn react link', () => {
  const { asFragment } = render(<App />, { wrapper: MemoryRouter });
  
  expect(asFragment()).toMatchSnapshot();
});
