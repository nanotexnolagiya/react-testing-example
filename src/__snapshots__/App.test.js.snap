// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <header
      class="App-header"
    >
      <div
        class="data-list"
      >
        <h1>
          Tasks
        </h1>
        <div
          class="input-wrap "
        >
          <label
            for="input-:r0:"
          >
            Add new task
          </label>
          <input
            id="input-:r0:"
            placeholder="New task title"
            type="text"
          />
        </div>
        Loading
      </div>
    </header>
    <div
      class="App-line"
    />
    <main
      class="App-main"
    >
      <img
        alt="logo"
        class="App-logo"
        src="logo.svg"
      />
      <a
        aria-current="page"
        class="App-link"
        href="/"
      >
        Tasks
      </a>
      <a
        href="/posts"
      >
        Posts
      </a>
    </main>
  </div>
</DocumentFragment>
`;
